### ISSUE TITLE
bookmark order is different after upgrade

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-ios#3006]
### Description: 
After upgrading from 1.20 to 1.21, @srirambv and I noticed that the order of some bookmark/folders in the UI can be different.

### Steps to Reproduce 

1. Have this file:
[ios copy.txt](https://github.com/brave/brave-ios/files/5475898/ios.copy.txt)
2. Rename to .html
3. Import on desktop version 
4. Using 1.20 on iOS, mimic this setup (order, folders, bookmarks, etc)
5.  Note how the "Sports" folder looks:

```
Sports (folder)
   espncricinfo.com (site)
   Cricket (folder)
   Sports (folder)
```

6. Upgrade to 1.21
7. Wait a minute 
8. Look at "Sports" folder now

```
Sports (folder)
   Cricket (folder)
   espncricinfo.com (site)
   Sports (folder)
```

**Actual result:** 
order is different
![IMG_0090](https://user-images.githubusercontent.com/28145373/97884426-0f58ae00-1cf4-11eb-9a57-2aee6ab67893.jpg)

**Expected result:**
order to be the same
![IMG_0091](https://user-images.githubusercontent.com/28145373/97884438-12539e80-1cf4-11eb-86ca-d8c609734f76.jpg)

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
easily

**Brave Version:** 
1.21 (20.10.30.21)

**Device details:** 
iPad iOS13

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

**Comment by iccub** [comment]
Should be fixed in `Brave Dev 1.21 (20.11.2.17)` please retest

**Comment by LaurenWags** [comment]
Verified passed with iOS13 iPad mini using `Brave Dev 1.21 (20.11.2.17)`

----------------
Verification passed on iPhone XR with iOS 13.5 running 1.21(20.11.2.20)
- Verified bookmarks/folders order is maintained after upgrade
----------------
Verification passed on iPhone 7+ with iOS 14.1 running 1.21(20.11.2.20)
- Verified bookmarks/folders order is maintained after upgrade
